providers/github.go

```diff
@@ -198,6 +206,43 @@ func (p *GitHubProvider) hasOrgAndTeam(accessToken string) (bool, error) {
 	return false, nil
 }
 
+// ValidateUser validates that the user exists as a collaborator on the specified repo
+func (p *GitHubProvider) ValidateUser(user string) bool {
+	//https://developer.github.com/v3/repos/collaborators/#check-if-a-user-is-a-collaborator
+
+	if p.Repo == "" {
+		return true
+	}
+
+	if p.OwnerToken == "" {
+		panic("OwnerToken must be provided to check repo collaboration")
+	}
+
+	return p.hasRepoCollaboration(user)
+}
+
+func (p *GitHubProvider) hasRepoCollaboration(user string) bool {
+	repos := strings.Split(p.Repo, ",")
+	for _, repo := range repos {
+
+		endpoint := &url.URL{
+			Scheme: p.ValidateURL.Scheme,
+			Host:   p.ValidateURL.Host,
+			Path:   path.Join(p.ValidateURL.Path, fmt.Sprintf("/repos/%s/collaborators/%s", repo, user)),
+		}
+		req, _ := http.NewRequest("GET", endpoint.String(), nil)
```

Comment:
I'm not a fan of ignoring errors, should we not capture and handle this error somehow? Should we default to closed if there is an issue to be safe?

Reply:
fixed